CommonClient.py

```diff
@@ -1037,11 +1037,12 @@ async def main(args):
 
     if args.url:
         url = urllib.parse.urlparse(args.url)
-        args.connect = url.netloc
-        if url.username:
-            args.name = urllib.parse.unquote(url.username)
-        if url.password:
-            args.password = urllib.parse.unquote(url.password)
+        if url.scheme == "archipelago":
+            args.connect = url.netloc
+            if url.username:
+                args.name = urllib.parse.unquote(url.username)
+            if url.password:
+                args.password = urllib.parse.unquote(url.password)
```

Comment:
this part is still needed, if that's rolled into a different PR or this PR is reopened removing the Launcher.py changes idc

Reply:
I'll reopen this and look at it soon, what exactly does this part do on its own?

Reply:
currently argparse grabs the first positional arg and throws it into url, that's to be expected but does mean that `launcher "text client"` puts the component name in the url block, this causes issues because the highlighted code block only checks if there is _a_ url arg before overriding things

the updated code just confirms that the scheme of the url is correct (i.e. `archipelago://blah@blah`) before applying any of the available values to the ctx to stop mishandling of any potential bad data